### PR TITLE
Only grant permissions if nicely asked DW5415

### PR DESCRIPTION
**Pull request by tea-short**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,10 @@
 # manage-mysql-user
 AWS Lambda function to manage MySQL users
+
+## Usage
+The Lambda accepts the following keys in the payload JSON:   
+* `mysql_user_username` - (Required) MySQL username whose password will be updated
+* `mysql_user_password_parameter_name` - (Optional, conflicts with `mysql_user_password_secret_name`) Name of SSM parameter that is used to store MySQL user's password
+* `mysql_user_password_secret_name` - (Optional, conflicts with `mysql_user_password_parameter_name`) Name of Secrets Manager secret that is used to store MySQL user's password
+* `privileges` - (Optional) MySQL privileges to grant. Valid values are `ALL`, `SELECT`. No privileges will be granted if empty or not supplied.
```
Comment (dan-hill2802) [inline README.md]:
```suggestion
* `privileges` - (Optional) MySQL privileges to grant (applied to entire database). Valid values are `ALL`, `SELECT`. No privileges will be granted if empty or not supplied.
```

File: manage-mysql-user.py [hunk 2]
```diff
@@ -424,12 +433,21 @@ def handler(event, context):
...
+                f"Granting {privileges} privileges to MySQL user {mysql_user_username}"
+            )
+            execute_statement(
+                "GRANT {} ON `{}`.* to '{}'@'%';".format(
+                    privileges, database, mysql_user_username
+                ),
+                mysql_master_username,
+                mysql_master_password_source,
+                mysql_master_password_source_type,
+            )
```
Comment (dan-hill2802) [inline manage-mysql-user.py:450]: might be worth a log to state "no privileges granted"

Comment (tea-short) [reply manage-mysql-user.py:450]: Good call, done

File: manage-mysql-user.py [hunk 3]
```diff
@@ -424,12 +433,25 @@ def handler(event, context):
...
-    )
+
+    if "privileges" in event.keys():
+        if len(event["privileges"]) > 0:
+            privileges = event["privileges"]
+            logger.info(
+                f"Granting {privileges} privileges to MySQL user {mysql_user_username}"
+            )
+            execute_statement(
+                "GRANT {} ON `{}`.* to '{}'@'%';".format(
```
Comment (dan-hill2802) [inline manage-mysql-user.py:444]: pretty sure that `if event["privileges"] not in ["ALL", "SELECT", ""]` on line 322 prevents a SQL injection here

Comment (tea-short) [reply manage-mysql-user.py:444]: well yes, that's the logic behind it

**Review by dan-hill2802 — APPROVED** [review]

**Pull request merged**